frontend/views/utils/markdown-utils.js

```diff
@@ -33,11 +33,21 @@ export function renderMarkdown (str: string): any {
   // There is some caveats discovered with 'dompurify' and DOMParser() API regarding how they interpret '<' and '>' characters.
   // So manually converting them to '&lt;' and '&gt;' here first.
   // ( context: https://github.com/okTurtles/group-income/issues/2130 )
-  str = str.replace(/</g, '&lt;').replace(/>/g, '&gt;')
+  const strSplitByCodeMarkdown = splitStringByMarkdownCode(str)
+  strSplitByCodeMarkdown.forEach((entry, index) => {
+    if (entry.type === 'plain' && strSplitByCodeMarkdown[index - 1]?.text !== '```') {
+      let entryText = entry.text
+      entryText = entryText.replace(/</g, '&lt;').replace(/>/g, '&gt;')
 
-  str = str.replace(/\n(?=\n)/g, '\n<br>')
-    .replace(/<br>\n(\s*)(\d+\.|-|```)/g, '\n\n$1$2') // custom-handling the case where <br> is directly followed by the start of block-code (```)
-    .replace(/(\d+\.|-)(\s.+)\n<br>/g, '$1$2\n\n') // this is a custom-logic added so that the end of ordered/un-ordered lists are correctly detected by markedjs.
+      entryText = entryText.replace(/\n(?=\n)/g, '\n<br>')
+        .replace(/<br>\n(\s*)(\d+\.|-)/g, '\n\n$1$2') // custom-handling the case where <br> is directly followed by the start of ordered/unordered lists
+        .replace(/(\d+\.|-)(\s.+)\n<br>/g, '$1$2\n\n') // this is a custom-logic added so that the end of ordered/un-ordered lists are correctly detected by markedjs.
+
+      entry.text = entryText
+    }
+  })
+
+  str = combineMarkdownSegmentListIntoString(strSplitByCodeMarkdown)
```

Comment:
The cause of the issue was that,
The additional fixes & adjustments related to white-spaces, line-breaks and `<` and `>` don't need to be applied to the contents of `<code>...</code>`. So updates have been here accordingly.